zerver/tests/test_decorators.py

```diff
@@ -736,3 +758,57 @@ def _login(self, user_email, password=None):
             user_profile.set_password(password)
             user_profile.save()
         self.login(user_email, password)
+
+class TestAuthenticatedJsonViewDecorator(ZulipTestCase):
+    def test_authenticated_json_view_if_wrong_subdomain(self):
+        user_email = 'hamlet@zulip.com'
+        self._login(user_email)
+        with self.settings(REALMS_HAVE_SUBDOMAINS=True):
+            self.assert_json_error_contains(self._do_test(user_email),
+                                            "Account is not associated with this "
+                                            "subdomain")
+
+    def _do_test(self, user_email):
+        data = {"status": '"started"'}
+        return self.client_post(r'/json/tutorial_status', data)
+
+    def _login(self, user_email, password=None):
+        if password:
+            user_profile = get_user_profile_by_email(user_email)
+            user_profile.set_password(password)
+            user_profile.save()
+        self.login(user_email, password)
+
+class TestLoggedInAndActive(ZulipTestCase):
+    def test_logged_in_and_active_if_wrong_subdomain(self):
+        rf = RequestFactory()
+        request = rf.get('/')
+        request.get_host = lambda: 'acme.testserver'
+        user_email = 'hamlet@zulip.com'
+        user_profile = get_user_profile_by_email(user_email)
+        request.user = user_profile
+
+        with self.settings(REALMS_HAVE_SUBDOMAINS=True):
+            result = logged_in_and_active(request)
+            self.assertFalse(result)
```

Comment:
It seems like it might be better to test the higher level `zulip_login_required` instead?